### PR TITLE
Implemented Gardevoir and Gardevoir Lv.X

### PR DESCRIPTION
**Pull request by Grayahm**

### CODE REVIEW

File: src/tcgwars/logic/impl/gen4/SecretWonders.groovy [hunk 1]
```diff
@@ -341,18 +341,33 @@ public enum SecretWonders implements LogicCardInfo {
           pokePower "Telepass", {
             text "Once during your turn , you may search your opponent’s discard pile for a Supporter card and use the effect of that card as the effect of this power. (This Supporter card remains in you opponent’s discard pile.) You can’t use more than 1 Telepass Poké-Power each turn. This power can’t be used if Gardevoir is affected by a Special Condition."
             actionA {
-            }
-          }
+              checkLastTurn()
+              assert bg.em().retrieveObject("Telepass") != bg.turnCount : "You can't use more than 1 Telepass Poke-Power each turn."
+              assert opp.discard.hasType(SUPPORTER) : "Your opponent has no supporters discarded."
+              if(opp.discard.hasType(SUPPORTER)){
```
Comment (flagrama) [inline src/tcgwars/logic/impl/gen4/SecretWonders.groovy:347]: This if statement is redundant due to the assert above

Comment (Grayahm) [reply src/tcgwars/logic/impl/gen4/SecretWonders.groovy]: fixed!

File: src/tcgwars/logic/impl/gen4/SecretWonders.groovy [hunk 2]
```diff
@@ -341,18 +341,33 @@ public enum SecretWonders implements LogicCardInfo {
           pokePower "Telepass", {
             text "Once during your turn , you may search your opponent’s discard pile for a Supporter card and use the effect of that card as the effect of this power. (This Supporter card remains in you opponent’s discard pile.) You can’t use more than 1 Telepass Poké-Power each turn. This power can’t be used if Gardevoir is affected by a Special Condition."
             actionA {
-            }
-          }
+              checkLastTurn()
+              assert bg.em().retrieveObject("Telepass") != bg.turnCount : "You can't use more than 1 Telepass Poke-Power each turn."
```
Comment (flagrama) [inline src/tcgwars/logic/impl/gen4/SecretWonders.groovy:345]: You need to store this object once the pokePower is used as well. As it is you will be retrieving a value that never equates to `bg.turnCount` no matter how many Gardevoir use the Poke Power.

File: src/tcgwars/logic/impl/gen4/SecretWonders.groovy [hunk 3]
```diff
@@ -2769,14 +2784,27 @@ public enum SecretWonders implements LogicCardInfo {
           pokePower "Teleportation", {
             text "Once during your turn , choose 1 of your Active Pokémon or 1 or your Benched Pokémon and switch Gardevoir with that Pokémon. This power can’t be used if Gardevoir is affected by a Special Condition."
             actionA {
```
Comment (flagrama) [inline src/tcgwars/logic/impl/gen4/SecretWonders.groovy:2784]: You need to check that Gardevoir doesn't have a status condition affecting it when using this power.

File: src/tcgwars/logic/impl/gen4/SecretWonders.groovy (hunk 2 above)
Comment (Grayahm) [reply src/tcgwars/logic/impl/gen4/SecretWonders.groovy:346]: Hopefully fixed

File: src/tcgwars/logic/impl/gen4/SecretWonders.groovy (hunk 3 above)
Comment (Grayahm) [reply src/tcgwars/logic/impl/gen4/SecretWonders.groovy:2786]: Fixed

File: src/tcgwars/logic/impl/gen4/SecretWonders.groovy (hunk 2 above)
Comment (flagrama) [reply src/tcgwars/logic/impl/gen4/SecretWonders.groovy:346]: There's actually a `checkNoSPC()` usually used for Pokémon Powers

File: src/tcgwars/logic/impl/gen3/FireRedLeafGreen.groovy [hunk 4]
```diff
@@ -449,6 +449,7 @@ public enum FireRedLeafGreen implements LogicCardInfo {
             text "Once during your turn (before your attack), you may choose any 1 card from your deck and put it into your hand. Shuffle your deck afterward. You can’t use more than 1 Quick Search Poké-Power each turn. This power can’t be used if Pidgeot is affected by a Special Condition."
             actionA {
               checkNoSPC()
+              assert !(self.specialConditions) : "Pidgeot is affected by a Special Condition."
```
Comment (flagrama) [inline src/tcgwars/logic/impl/gen3/FireRedLeafGreen.groovy:452]: This one is completely unnecessary due to the line above.

File: src/tcgwars/logic/impl/gen4/SecretWonders.groovy (hunk 2 above)
Comment (flagrama) [reply src/tcgwars/logic/impl/gen4/SecretWonders.groovy:346]: Also, should probably unindent some of this now that it isn't in an if block.

File: src/tcgwars/logic/impl/gen4/SecretWonders.groovy [hunk 5]
```diff
@@ -341,18 +341,33 @@ public enum SecretWonders implements LogicCardInfo {
...
+          } 
           move "Psychic Lock", {
             text "60 damage. During your opponent’s next turn, your opponent can’t use any Poké-Powers on his or her Pokémon."
             energyCost P, C, C
             attackRequirement {}
             onAttack {
-              damage 0
+              damage 60
+              getterA (IS_ABILITY_BLOCKED) { Holder h ->
+              if (h.effect.ability instanceof PokePower) {
```
Comment (luongthomas) [inline src/tcgwars/logic/impl/gen4/SecretWonders.groovy:363]: Could you indent the `if` block over to the right by one tab?

File: src/tcgwars/logic/impl/gen3/FireRedLeafGreen.groovy [hunk 6]
```diff
@@ -2523,6 +2524,7 @@ public enum FireRedLeafGreen implements LogicCardInfo {
           pokePower "Extra Energy Bomb", {
             text "Once during your turn (before your attack), you may discard Electrode ex and all the cards attached to it (this counts as Knocking Out Electrode ex). If you do, search your discard pile for 5 Energy cards and attach them to any of your Pokémon (excluding Pokémon-ex) in any way you like. This power can’t be used if Electrode ex is affected by a Special Condition."
             actionA {
+              assert !(self.specialConditions) : "Electrode ex is affected by a Special Condition."
```
Comment (luongthomas) [inline src/tcgwars/logic/impl/gen3/FireRedLeafGreen.groovy:2527]: You can swap this out to `checkNoSPC()`

File: src/tcgwars/logic/impl/gen4/SecretWonders.groovy (hunk 3 above)
Comment (luongthomas) [reply src/tcgwars/logic/impl/gen4/SecretWonders.groovy:2786]: Looks like `checkNoSPC()` is still missing from this ability

File: src/tcgwars/logic/impl/gen3/FireRedLeafGreen.groovy (hunk 4 above)
Comment (Grayahm) [reply src/tcgwars/logic/impl/gen3/FireRedLeafGreen.groovy]: Yep removing all those now

File: src/tcgwars/logic/impl/gen4/SecretWonders.groovy (hunk 5 above)
Comment (Grayahm) [reply src/tcgwars/logic/impl/gen4/SecretWonders.groovy]: fixed

**Pull request merged**